manifests/install.pp

```diff
@@ -268,25 +268,34 @@
         }
       }
 
-      if "${::python::version}" =~ /^python3/ { #lint:ignore:only_variable_string
+      if String($::python::version) =~ /^python3/ {
         $pip_category = undef
-        $pip_package = 'python3-pip'
+        $pip_package = "${python}-pip"
+        $pip_provider = $python.regsubst(/^.*python3\.?/,'pip3.').regsubst(/\.$/,'')
       } elsif ($::osfamily == 'RedHat') and (versioncmp($::operatingsystemmajrelease, '7') >= 0) {
         $pip_category = undef
         $pip_package = 'python2-pip'
-      } elsif $::osfamily == 'Gentoo' {
-        $pip_category = 'dev-python'
-        $pip_package = 'pip'
+        $pip_provider = pip2
       } else {
-        $pip_category = undef
-        $pip_package = 'python-pip'
+        if $::osfamily == 'Gentoo' {
```

Comment:
Can this be kept as an elsif block, like the others? The nesting inside the final else block is a little confusing.

Reply:
It could, but then the two assignments get repeated unnecessarily.